apps/browser/src/autofill/services/autofill-overlay-content.service.ts

```diff
@@ -1313,13 +1358,34 @@ export class AutofillOverlayContentService implements AutofillOverlayContentServ
   }
 
   /**
-   * Updates the local reference to the inline menu visibility setting.
+   * Updates the local reference to the passed inline menu visibility setting.
    *
    * @param data - The data object from the extension message.
    */
   private updateInlineMenuVisibility({ data }: AutofillExtensionMessage) {
-    if (!isNaN(data?.inlineMenuVisibility)) {
-      this.inlineMenuVisibility = data.inlineMenuVisibility;
+    const settingNewValue = data?.newSettingValue;
+
+    if (settingNewValue == null) {
+      return;
+    }
+
+    const settingType = data?.settingType;
+
+    // Setting value update is for overall inline menu visibility
+    if (settingType == null && !isNaN(settingNewValue as InlineMenuVisibilitySetting)) {
+      this.inlineMenuVisibility = settingNewValue as InlineMenuVisibilitySetting;
+
+      return;
+    }
+
+    if (typeof settingType === "boolean") {
+      if (settingType === CipherType.Card) {
+        this.showInlineMenuCards = settingNewValue as boolean;
+      }
+
+      if (settingType === CipherType.Identity) {
+        this.showInlineMenuIdentities = settingNewValue as boolean;
+      }
```

Comment:
```suggestion
    if (!isNaN(data?.inlineMenuVisibility)) {
      this.inlineMenuVisibility = data.inlineMenuVisibility;
```

Suggestion - Revert this, in effect the script reload that is happening when the inline menu cards and inline menu identities setting is changed will handle the expected change to happen on the content script side.

Reply:
4a355b2f085c0e2db21a4286a45218b192239fab